### PR TITLE
fix(): fix multiple document not working when using fastify

### PR DESCRIPTION
**Pull request by Boshen**
fixes https://github.com/nestjs/swagger/issues/234
fixes https://github.com/fastify/fastify-swagger/issues/213

## PR Checklist
Please check if your PR fulfills the following requirements:

- [*] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[*] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[*] No
```

## Other information

### CODE REVIEW

**Comment by Boshen** [comment]
see https://github.com/fastify/fastify-swagger/pull/216#pullrequestreview-328728548

File: lib/swagger-module.ts [hunk 1]
```diff
@@ -65,19 +65,21 @@ export class SwaggerModule {
...
-        require('fastify-swagger')
-      ),
-      {
-        swagger: document,
-        exposeRoute: true,
-        routePrefix: path,
-        mode: 'static',
-        specification: {
-          document
+    httpServer.register(async (httpServer: any) => {
```
Comment (nartc) [inline lib/swagger-module.ts]: what happens here?

**Comment by kamilmysliwiec** [comment]
Can you create an integration test for this change too?

File: lib/swagger-module.ts (hunk 1 above)
Comment (Boshen) [reply lib/swagger-module.ts]: Apparently fastify plugin works differently. Per its [documentation](https://www.fastify.io/docs/latest/Plugins/), this adds a different scope so the internal call
`fastify.decorate('swagger', swagger)` does not collide when we try to make multiple swagger docs

**Comment by Boshen** [comment]
> Can you create an integration test for this change too?

@kamilmysliwiec Done!

**Comment by Boshen** [comment]
@kamilmysliwiec can you clean this up and merge this? I really want this for my project, thanks!

**Comment by kamilmysliwiec** [comment]
Thank you @Boshen!

**Pull request merged**